rmw_microxrcedds_c/src/rmw_microros_internal/error_handling_internal.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ */
+
+#ifndef RMW_MICROROS_INTERNAL__ERROR_HANDLING_INTERNAL_H_
+#define RMW_MICROROS_INTERNAL__ERROR_HANDLING_INTERNAL_H_
+
+#include <rmw/error_handling.h>
+#include <rmw_microxrcedds_c/config.h>
+#include <rmw_microros/error_handling.h>
+
+#if defined(__cplusplus)
+extern "C"
+{
+#endif  // if defined(__cplusplus)
+
+#ifdef RMW_UROS_ERROR_HANDLING
+extern rmw_uros_error_handling error_callback;
+
+#define RMW_UROS_TRACE_ERROR(entity, source, desc, ...) \
+  RMW_SET_ERROR_MSG(desc); \
+  if (NULL != error_callback) { \
+    rmw_uros_error_context_t ctx = {__VA_ARGS__}; \
+    ctx.description = desc; \
+    error_callback(entity, source, ctx, __FILE__, __LINE__); \
```

Comment:
```suggestion
    error_callback(source, entity, ctx, __FILE__, __LINE__); \
```

Argument order is different from error_callback:
https://github.com/micro-ROS/rmw_microxrcedds/blob/200fca2688d5a620ad32a4540abe6920f9ed8945/rmw_microxrcedds_c/include/rmw_microros/error_handling.h#L69-L74

Reply:
```suggestion
    error_callback(entity, source, ctx, __FILE__, __LINE__); \
    rmw_reset_error(); \
```
Should we clear the error state after its been notified? 
Otherwise we will get messages like this for multiple errors::

```
>>> [rcutils|error_handling.c:108] rcutils_set_error_state()
This error state is being overwritten:

  'node handle is null, at /uros_ws/src/uros/rmw_microxrcedds/rmw_microxrcedds_c/src/rmw_init.c:118'

with this new error message:

  'Not available static buffer memory node in on_topic callback, at /uros_ws/src/uros/rmw_microxrcedds/rmw_microxrcedds_c/src/callbacks.c:79'

rcutils_reset_error() should be called after error handling to avoid this.
<<<
```

Reply:
solved in e0d989e377429b5756ccdda8534ef4d84f1b4b67